go-controller/pkg/networkmanager/nad_controller.go

```diff
@@ -331,7 +411,43 @@ func (c *nadController) syncNAD(key string, nad *nettypes.NetworkAttachmentDefin
 		if len(oldNetwork.GetNADs()) == 0 {
 			c.networkController.DeleteNetwork(oldNetworkName)
 		} else {
-			c.networkController.EnsureNetwork(oldNetwork)
+			filtered := false
+			if config.OVNKubernetesFeature.EnableDynamicUDNAllocation && c.nadFilterFunc != nil {
```

Comment:
I would probably rather only assign `c.nadFilterFunc` on init if the feature is enabled , and then only check `if c.nadFilterFunc != nil` later, as it seems easy to forget to check this flag